crates/snapbox/src/assert.rs

```diff
@@ -197,17 +195,20 @@ impl Assert {
         expected: crate::Result<crate::Data>,
         mut actual: crate::Data,
     ) -> (crate::Result<crate::Data>, crate::Data) {
-        let expected = expected.map(|d| d.map_text(crate::utils::normalize_lines));
+        let expected = expected.map(|d| d.normalize(NormalizeNewlines));
         // On `expected` being an error, make a best guess
-        if let (Some(expected), format) = expected
-            .as_ref()
-            .map(|d| (d.as_str(), d.format()))
-            .unwrap_or((Some(""), DataFormat::Text))
-        {
+        let ok = expected.clone().unwrap_or_default();
+
+        if self.normalize_paths {
+            actual = actual
+                .try_coerce(ok.format())
+                .normalize(NormalizeText)
+                .normalize(NormalizeMatches::new(&self.substitutions, ok));
+        } else {
             actual = actual
-                .try_coerce(format)
-                .map_text(|s| self.normalize_text(s))
-                .map_text(|t| self.substitutions.normalize(t, expected));
+                .try_coerce(ok.format())
+                .normalize(NormalizeNewlines)
+                .normalize(NormalizeMatches::new(&self.substitutions, ok));
         }
```

Comment:
This doesn't do a good job communicating what is happening.  What if instead we just dropped `NormalizeText` and had all callers of it call the parts its made up of?